core/include/vecmem/containers/impl/device_array.ipp

```diff
@@ -1,11 +1,16 @@
 /* VecMem project, part of the ACTS project (R&D line)
  *
- * (c) 2021 CERN for the benefit of the ACTS project
+ * (c) 2021-2024 CERN for the benefit of the ACTS project
  *
  * Mozilla Public License Version 2.0
  */
 #pragma once
 
+// HIP include(s)
+#if defined(__HIP_DEVICE_COMPILE__)
+#include <hip/hip_runtime.h>
```

Comment:
This was a good catch! :thinking:

Though now I fear that we're missing such statements in a bunch of other `.ipp` files as well. :frowning: (Whatever is used on the device, and uses `assert(...)`. Which is a lot of this project's code.)

But well, the tests of the project now build successfully, so let's just stay with this for now.